File: src/ChainMetrics.UI/app/types/order.py
from datetime import date
from pydantic import BaseModel, Field


class Order(BaseModel):
    id: str = Field(..., alias="id")
    customer_id: str = Field(..., alias="customerId")
    status: str = Field(..., alias="status")
    purchase_timestamp: date = Field(..., alias="purchaseTimestamp")
    approved_at: date = Field(..., alias="approvedAt")
    delivered_timestamp: date = Field(..., alias="deliveredTimestamp")
    estimated_delivery_date: date = Field(..., alias="estimatedDeliveryDate")

class PerStatusDTO(BaseModel):
    status: str = Field(..., alias="status")
    quantity: int = Field(..., alias="quantity")
    total_delivery_hours: float = Field(..., alias="totalDeliveryHours")

File: src/ChainMetrics.Domain/Entities/Order.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ChainMetrics.Domain.Entities;

[Table("df_Orders")]
public class Order
{
    [Key]
    [Column("order_id")]
    public required string Id { get; set; }

    [Column("customer_id")]
    public required string CustomerId { get; set; }
    public Customer? Customer { get; set; }

    [Column("order_status")]
    public required string Status { get; set; }

    [Column("order_purchase_timestamp")]
    public DateTime PurchaseTimestamp { get; set; }

    [Column("order_approved_at")]
    public DateTime? ApprovedAt { get; set; }

    [Column("order_delivered_timestamp")]
    public DateTime? DeliveredTimestamp { get; set; }

    [Column("order_estimated_delivery_date")]
    public DateOnly? EstimatedDeliveryDate { get; set; }
}
